File: examples/javascript/network-tools.js
/**
 * Devsly API - Network Tools Examples (JavaScript/Node.js)
 *
 * Requirements: Node.js 12+, axios
 * Install: npm install axios
 *
 * Get your API key from: https://devsly.io/dashboard
 */

const axios = require('axios');

class DevslyNetworkTools {
    constructor(apiKey) {
        this.apiKey = apiKey;
        this.baseUrl = 'https://devsly.io/api/v1';
        this.headers = {
            'X-API-Key': apiKey,
            'Content-Type': 'application/json'
        };
    }

    /**
     * Make API request
     */
    async request(endpoint, params = {}) {
        try {
            const response = await axios.get(`${this.baseUrl}${endpoint}`, {
                headers: this.headers,
                params: params
            });
            return response.data;
        } catch (error) {
            throw new Error(`API request failed: ${error.message}`);
        }
    }

    /**
     * WHOIS Lookup
     *
     * @param {string} domain - Domain name to lookup
     * @returns {Promise<Object>} WHOIS information
     */
    async whoisLookup(domain) {
        return this.request('/network/whois', { domain });
    }

    /**
     * DNS Lookup
     *
     * @param {string} domain - Domain name to lookup
     * @param {string} type - Record type (A, AAAA, MX, NS, TXT, CNAME, SOA)
     * @returns {Promise<Object>} DNS records
     */
    async dnsLookup(domain, type = 'A') {
        return this.request('/network/dns-lookup', { domain, type });
    }

    /**
     * IP Information
     *
     * @param {string|null} ip - IP address (null for current IP)
     * @returns {Promise<Object>} IP information
     */
    async ipInfo(ip = null) {
        const params = ip ? { ip } : {};
        return this.request('/network/ip-info', params);
    }
}

// Example Usage
(async () => {
    const api = new DevslyNetworkTools('YOUR_API_KEY');

    try {
        // WHOIS Lookup
        console.log('WHOIS Lookup for google.com:');
        const whois = await api.whoisLookup('google.com');
        console.log(`Registrar: ${whois.registrar || 'N/A'}`);
        console.log(`Creation Date: ${whois.creation_date || 'N/A'}`);
        console.log();

        // DNS Lookup
        console.log('DNS Lookup for github.com (A records):');
        const dns = await api.dnsLookup('github.com', 'A');
        if (dns.success && dns.records) {
            dns.records.forEach(record => console.log(`  ${record}`));
        }
        console.log();

        // DNS MX Records
        console.log('DNS Lookup for gmail.com (MX records):');
        const mx = await api.dnsLookup('gmail.com', 'MX');
        if (mx.success && mx.records) {
            mx.records.forEach(record => console.log(`  ${record}`));
        }
        console.log();

        // IP Information
        console.log('IP Information for 8.8.8.8:');
        const ipInfo = await api.ipInfo('8.8.8.8');
        console.log(`IP: ${ipInfo.ip || 'N/A'}`);
        console.log(`Country: ${ipInfo.country || 'N/A'}`);
        console.log(`City: ${ipInfo.city || 'N/A'}`);
        console.log(`ISP: ${ipInfo.isp || 'N/A'}`);
        console.log();

        // Current IP Information
        console.log('Current IP Information:');
        const current = await api.ipInfo();
        console.log(`Your IP: ${current.ip || 'N/A'}`);
        console.log(`Country: ${current.country || 'N/A'}`);

    } catch (error) {
        console.error('Error:', error.message);
    }
})();

// Export for use in other modules
module.exports = DevslyNetworkTools;

File: examples/python/network_tools.py
"""
Devsly API - Network Tools Examples (Python)

Requirements: Python 3.6+, requests library
Install: pip install requests

Get your API key from: https://devsly.io/dashboard
"""

import requests
from typing import Optional, Dict, Any


class DevslyNetworkTools:
    """Devsly Network Tools API Client"""

    def __init__(self, api_key: str):
        self.api_key = api_key
        self.base_url = "https://devsly.io/api/v1"
        self.headers = {
            "X-API-Key": api_key,
            "Content-Type": "application/json"
        }

    def _request(self, endpoint: str, params: Optional[Dict] = None) -> Dict[Any, Any]:
        """Make API request"""
        url = f"{self.base_url}{endpoint}"

        response = requests.get(url, headers=self.headers, params=params)
        response.raise_for_status()

        return response.json()

    def whois_lookup(self, domain: str) -> Dict[Any, Any]:
        """
        WHOIS Lookup

        Args:
            domain: Domain name to lookup

        Returns:
            WHOIS information
        """
        return self._request("/network/whois", {"domain": domain})

    def dns_lookup(self, domain: str, record_type: str = "A") -> Dict[Any, Any]:
        """
        DNS Lookup

        Args:
            domain: Domain name to lookup
            record_type: Record type (A, AAAA, MX, NS, TXT, CNAME, SOA)

        Returns:
            DNS records
        """
        return self._request("/network/dns-lookup", {
            "domain": domain,
            "type": record_type
        })

    def ip_info(self, ip: Optional[str] = None) -> Dict[Any, Any]:
        """
        IP Information

        Args:
            ip: IP address (None for current IP)

        Returns:
            IP information
        """
        params = {"ip": ip} if ip else {}
        return self._request("/network/ip-info", params)


# Example Usage
if __name__ == "__main__":
    # Initialize API client
    api = DevslyNetworkTools("YOUR_API_KEY")

    try:
        # WHOIS Lookup
        print("WHOIS Lookup for google.com:")
        whois = api.whois_lookup("google.com")
        print(f"Registrar: {whois.get('registrar', 'N/A')}")
        print(f"Creation Date: {whois.get('creation_date', 'N/A')}")
        print()

        # DNS Lookup
        print("DNS Lookup for github.com (A records):")
        dns = api.dns_lookup("github.com", "A")
        if dns.get("success"):
            for record in dns.get("records", []):
                print(f"  {record}")
        print()

        # DNS MX Records
        print("DNS Lookup for gmail.com (MX records):")
        mx = api.dns_lookup("gmail.com", "MX")
        if mx.get("success"):
            for record in mx.get("records", []):
                print(f"  {record}")
        print()

        # IP Information
        print("IP Information for 8.8.8.8:")
        ip_info = api.ip_info("8.8.8.8")
        print(f"IP: {ip_info.get('ip', 'N/A')}")
        print(f"Country: {ip_info.get('country', 'N/A')}")
        print(f"City: {ip_info.get('city', 'N/A')}")
        print(f"ISP: {ip_info.get('isp', 'N/A')}")
        print()

        # Current IP Information
        print("Current IP Information:")
        current = api.ip_info()
        print(f"Your IP: {current.get('ip', 'N/A')}")
        print(f"Country: {current.get('country', 'N/A')}")

    except requests.exceptions.RequestException as e:
        print(f"Error: {e}")
